src/components/ArticleDetail.jsx

```diff
@@ -0,0 +1,35 @@
+import { useParams, Link } from 'react-router-dom';
+import useFetch from '../hooks/useFetch';
+import '../assets/ArticleDetail.css';
+
+const ArticleDetail = () => {
+    const { articleId } = useParams();
+    const { data, isLoading, error } = useFetch(`https://nc-news-be-project-k6p0.onrender.com/api/articles/${articleId}`);
```

Comment:
THis is called useFetch but the file actually uses Axios. Ideally We would have something simialr to this but extracted to and api.js file where you export functions that deal with requests - one for each endpoint - that you can import into your components. The notes have a sections on this